snapcraft/tests/test_plugin_catkin.py

```diff
@@ -29,16 +30,67 @@ class _IOError(IOError):
     errno = os.errno.EACCES
 
 
-class CatkinTestCase(tests.TestCase):
+class CatkinPluginTestCase(tests.TestCase):
 
     def setUp(self):
         super().setUp()
 
         class props:
+            rosversion = 'foo'
             catkin_packages = ['my_package']
 
         self.properties = props()
 
+        patcher = mock.patch('snapcraft.repo.Ubuntu')
+        self.ubuntu_mock = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_pull_debian_dependencies(self):
+        package_xml = """<?xml version="1.0"?>
+                      <package>
+                        <buildtool_depend>buildtool_depend</buildtool_depend>
+                        <build_depend>build_depend</build_depend>
+                        <run_depend>run_depend</run_depend>
+                      </package>"""
+
+        plugin = catkin.CatkinPlugin('test-part', self.properties)
+
+        with mock.patch.object(builtins, 'open',
+                               mock.mock_open(read_data=package_xml)):
```

Comment:
In unit tests, I prefer to access the filesystem in temp directories instead of mocking that part.
But your test is correct, if you prefer it this way, I'm ok with that.